src/dotnet-watch/CommandLineOptions.cs

```diff
@@ -53,6 +53,10 @@ public static CommandLineOptions Parse(string[] args, IConsole console)
   DOTNET_WATCH
   dotnet-watch sets this variable to '1' on all child processes launched.
 
+  DOTNET_WATCH_ITERATION
+  dotnet-watch sets this variable to '0' and increments by one each time
```

Comment:
Perhaps 1 is a better starting count?

Reply:
I'm fine either way. starting at 1 was my first choice too, but then I probably overthought this on account of all those "arrays start at 1" jokes on Reddit. I've updated the PR to start the counter at 1.